bson/src/test/unit/org/bson/json/JsonReaderTest.java

```diff
@@ -967,7 +967,80 @@ public void testUuid() {
             assertEquals(AbstractBsonReader.State.DONE, bsonReader.getState());
             return null;
         });
+
+        // wrong length
+        json = "{ \"$uuid\" : \"b5f21e0c2-a0d-42d6-ad03-d827008d8ab\"}}";
+        testStringAndStream(json, bsonReader -> {
+            try {
+                bsonReader.readBinaryData();
+                fail("Should fail to parse NumberDecimal constructor with a string");
+            } catch (JsonParseException e) {
+                // all good
```

Comment:
Do we want to consider writing a separate test for each scenario with `expected = JsonParseException.class`? For example
```java
@Test(expected = JsonParseException.class)
public void testUuid_wrongLength() {
    testStringAndStream("{ \"$uuid\" : \"b5f21e0c2-a0d-42d6-ad03-d827008d8ab\"}}", bsonReader -> {
        bsonReader.readBinaryData();
        return null;
    });
}
```

Reply:
I considered it, but decided against it.  But I don't feel too strongly one way or the other.

Reply:
I ended up splitting out the validation to a separate helper class, and wrote table-driven tests of that class.